toonz/sources/toonz/comboviewerpane.cpp

```diff
@@ -413,13 +411,38 @@ void ComboViewerPanel::showEvent(QShowEvent *event) {
 
 void ComboViewerPanel::hideEvent(QHideEvent *event) {
   StyleShortcutSwitchablePanel::hideEvent(event);
-  TApp *app = TApp::instance();
-  disconnect(app->getCurrentScene());
-  disconnect(app->getCurrentLevel());
-  disconnect(app->getCurrentFrame());
-  disconnect(app->getCurrentObject());
-  disconnect(app->getCurrentXsheet());
-  disconnect(app->getCurrentTool());
+  TApp *app                    = TApp::instance();
+  TFrameHandle *frameHandle    = app->getCurrentFrame();
+  TSceneHandle *sceneHandle    = app->getCurrentScene();
+  TXshLevelHandle *levelHandle = app->getCurrentLevel();
+  TObjectHandle *objectHandle  = app->getCurrentObject();
+  TXsheetHandle *xshHandle     = app->getCurrentXsheet();
+
+  disconnect(xshHandle, SIGNAL(xsheetChanged()), this, SLOT(onSceneChanged()));
+
+  disconnect(sceneHandle, SIGNAL(sceneChanged()), this, SLOT(onSceneChanged()));
+  disconnect(sceneHandle, SIGNAL(nameSceneChanged()), this,
+             SLOT(changeWindowTitle()));
+  disconnect(sceneHandle, SIGNAL(sceneSwitched()), this,
+             SLOT(onSceneChanged()));
+  disconnect(levelHandle, SIGNAL(xshLevelSwitched(TXshLevel *)), this,
+             SLOT(onXshLevelSwitched(TXshLevel *)));
+  disconnect(levelHandle, SIGNAL(xshLevelChanged()), this,
+             SLOT(changeWindowTitle()));
+  disconnect(levelHandle, SIGNAL(xshLevelTitleChanged()), this,
+             SLOT(changeWindowTitle()));
+  disconnect(levelHandle, SIGNAL(xshLevelChanged()), this,
+             SLOT(updateFrameRange()));
+
+  disconnect(frameHandle, SIGNAL(frameSwitched()), this,
+             SLOT(changeWindowTitle()));
+  disconnect(frameHandle, SIGNAL(frameSwitched()), this,
+             SLOT(onFrameChanged()));
+  disconnect(frameHandle, SIGNAL(frameTypeChanged()), this,
+             SLOT(onFrameTypeChanged()));
```

Comment:
Thank you so much for fixing the bug!

`disconnect(app->getCurrentScene());` was obviously wrong as  `app->getCurrentScene()` is a sender, not a receiver...

How about to replacing lines 415-442 with the following lines for simplification? :
`disconnect(app->getCurrentFrame(), 0, this, 0);`
`disconnect(app->getCurrentScene(), 0, this, 0);`
`disconnect(app->getCurrentLevel(), 0, this, 0);`
`disconnect(app->getCurrentObject(), 0, this, 0);`
`disconnect(app->getCurrentXsheet(), 0, this, 0);`

According to the [Qt Documentation](http://doc.qt.io/qt-5/qobject.html#disconnect); 
> 0 may be used as a wildcard, meaning "any signal", "any receiving object", or "any slot in the receiving object", respectively.
>The sender may never be 0. (You cannot disconnect signals from more than one object in a single call.)
>If signal is 0, it disconnects receiver and method from any signal. If not, only the specified signal is disconnected.
>If receiver is 0, it disconnects anything connected to signal. If not, slots in objects other than receiver are not disconnected.
>If method is 0, it disconnects anything that is connected to receiver. If not, only slots named method will be disconnected, and all other slots are left alone. The method must be 0 if receiver is left out, so you cannot disconnect a specifically-named slot on all objects.